### ISSUE TITLE
ValueError in process()

### ISSUE THREAD

**Issue by Nithanaroy** [annoviko/pyclustering#438]
When running cure_instance.process() get a value error with ccore set to False. Runs fine with ccore set to True.
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-95-1d35447ead74> in <module>()
      3 k = 4
      4 cure_instance = cure(X, k, ccore=False);
----> 5 cure_instance.process();
      6 clusters = cure_instance.get_clusters();

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/cluster/cure.py in process(self)
    169                 self.__queue.remove(cluster2);
    170 
--> 171                 self.__delete_represented_points(cluster1);
    172                 self.__delete_represented_points(cluster2);
    173 

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/cluster/cure.py in __delete_represented_points(self, cluster)
    361 
    362         for point in cluster.rep:
--> 363             self.__tree.remove(point, payload=cluster);
    364 
    365 

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/container/kdtree.py in remove(self, point, **kwargs)
    273         node_for_remove = None;
    274         if ('payload' in kwargs):
--> 275             node_for_remove = self.find_node_with_payload(point, kwargs['payload'], None);
    276         else:
    277             node_for_remove = self.find_node(point, None);

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/container/kdtree.py in find_node_with_payload(self, point, payload, cur_node)
    421 
    422         rule_search = lambda node, point=point, payload=payload: node.data == point and node.payload == payload;
--> 423         return self.__find_node_by_rule(point, rule_search, cur_node);
    424 
    425 

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/container/kdtree.py in __find_node_by_rule(self, point, search_rule, cur_node)
    394             if (cur_node.data[cur_node.disc] <= point[cur_node.disc]):
    395                 # Check if it's required node
--> 396                 if (search_rule(cur_node)):
    397                     req_node = cur_node;
    398                     break;

<MY_PROJECT_PATH>/conda_env/lib/python3.6/site-packages/pyclustering/container/kdtree.py in <lambda>(node, point, payload)
    420         """
    421 
--> 422         rule_search = lambda node, point=point, payload=payload: node.data == point and node.payload == payload;
    423         return self.__find_node_by_rule(point, rule_search, cur_node);
    424 

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Comment by annoviko** [comment]
@Nithanaroy, 

Thank you for your report! Could you please share data that is used for clustering to help to reproduce the issue? If it is a big to display it here, could you please send it to pyclustering@yandex.ru?

**Comment by annoviko** [comment]
@Nithanaroy ,

Root cause of the problem: input data is not 'list', most probably 'numpy.array'. The last does not provide explicit overloading for __eq__ operator that returns boolean value, instead it returns array of values. As a result, ValueError exception is thrown by the comparison. I will provide correction for the problem.

**Comment by annoviko** [comment]
@Nithanaroy,

The correction has been delivered to branches '0.8.dev' and 'master'. It will be available in the next release 0.8.2. Thank you again for the bug reporting.
